File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.kt
class Solution {
    fun minOperations(nums: IntArray): Int {
        var res = 0
        for (i in 0..(nums.size-2)) {
            if (nums[i] < nums[i+1]) continue
            val diff = Math.abs(nums[i]-nums[i+1]) + 1
            nums[i+1] += diff
            res += diff
        }
        return res
    }
}

File: 1827-minimum-operations-to-make-the-array-increasing/1827-minimum-operations-to-make-the-array-increasing.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        res = 0
        for i in range(1, len(nums)):
            if nums[i-1] < nums[i]: 
                continue
            diff = abs(nums[i] - nums[i-1]) + 1
            nums[i] += diff
            res += diff
        return res